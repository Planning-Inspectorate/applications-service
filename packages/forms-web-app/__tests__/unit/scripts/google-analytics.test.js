/**
 * @jest-environment jsdom
 */
/* eslint-env browser */

const { initialiseGoogleAnalytics } = require('../../../src/scripts/google-analytics');

describe('scripts/google-analytics', () => {
	const fakeGaId = 'some-test-value';
	const gaIdElement = document.createElement('p');
	gaIdElement.id = 'gaId';
	gaIdElement.textContent = fakeGaId;

	const setupFakeDom = () => {
		document.body.append(gaIdElement);
		const script = document.createElement('script');
		document.body.appendChild(script);
	};

	beforeEach(() => {
		jest.resetAllMocks();

		setupFakeDom();
	});

	afterEach(() => {
		jest.useRealTimers();
		document.body.innerHTML = '';
	});

	it('should initialise google analytics script', () => {
		initialiseGoogleAnalytics(document);

		expect(document.body).toMatchSnapshot();
	});

	it('should create a script tag with the correct src attribute', () => {
		initialiseGoogleAnalytics(document);

		const scriptTags = document.getElementsByTagName('script');
		const gaScriptTag = Array.from(scriptTags).find(
			(script) => script.src === `https://www.googletagmanager.com/gtag/js?id=${fakeGaId}`
		);

		expect(gaScriptTag).toBeDefined();
		expect(gaScriptTag.async).toBe(true);
		expect(gaScriptTag.type).toBe('text/javascript');
	});

	it('should not create a script tag if gaId is not present', () => {
		document.getElementById('gaId').remove();

		initialiseGoogleAnalytics(document);

		const scriptTags = document.getElementsByTagName('script');
		const gaScriptTag = Array.from(scriptTags).find(
			(script) => script.src === `https://www.googletagmanager.com/gtag/js?id=${fakeGaId}`
		);

		expect(gaScriptTag).toBeUndefined();
	});
});
