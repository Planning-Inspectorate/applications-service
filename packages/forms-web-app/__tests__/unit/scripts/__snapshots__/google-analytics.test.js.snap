// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scripts/google-analytics should initialise google analytics script 1`] = `
<body>
  <p
    id="gaId"
  >
    some-test-value
  </p>
  <script
    src="https://www.googletagmanager.com/gtag/js?id=some-test-value"
    type="text/javascript"
  />
  <script />
</body>
`;
